kyo-data/shared/src/test/scala/kyo/RecordTest.scala

```diff
@@ -0,0 +1,469 @@
+
+            def takesNameOnly(r: Record["name" ~ String]): String =
+                s"Name: ${r("name")}"
+
+            assert(takesNameAge(record) == "Bob is 25")
+            assert(takesNameOnly(record) == "Name: Bob")
+        }
+    }
+
+    "equality and hashCode" - {
+        // Bypass CanEqual since it's not provided
+        given [A, B]: CanEqual[A, B] = CanEqual.derived
+
+        "equal records with same fields" in {
+            val record1 = "name" ~ "Alice" & "age" ~ 30
+            val record2 = "name" ~ "Alice" & "age" ~ 30
+
+            assert(record1 == record2)
+            assert(record1.hashCode == record2.hashCode)
+        }
+
+        "not equal with different values" in {
+            val record1 = "name" ~ "Alice" & "age" ~ 30
+            val record2 = "name" ~ "Alice" & "age" ~ 31
+
+            assert(record1 != record2)
+        }
+
+        "not equal with different field names" in {
+            val record1 = "name" ~ "Alice" & "age" ~ 30
+            val record2 = "name" ~ "Alice" & "years" ~ 30
+
+            assert(record1 != record2)
+        }
+
+        "equal with fields in different order" in {
+            val record1 = "name" ~ "Alice" & "age" ~ 30
+            val record2 = "age" ~ 30 & "name" ~ "Alice"
+
+            assert(record1 == record2)
+            assert(record1.hashCode == record2.hashCode)
+        }
+
+        "not equal with different number of fields" in {
+            val record1 = "name" ~ "Alice" & "age" ~ 30
+            val record2 = "name" ~ "Alice"
+
+            assert(record1 != record2)
+        }
+    }
+
+    "duplicate field names" - {
+        "allows fields with same name but different types" in {
+            val record = "value" ~ "string" & "value" ~ 42
+
+            assert(record[String]("value") == "string")
+            assert(record[Int]("value") == 42)
+        }
+
+        "preserves both values when merging records" in {
+            val record1 = "value" ~ "string"
+            val record2 = "value" ~ 42
+            val merged  = record1 & record2
+
+            assert(merged[String]("value") == "string")
+            assert(merged[Int]("value") == 42)
+        }
+
+        "handles multiple duplicate fields" in {
+            val record = "x" ~ "str" & "x" ~ 42 & "x" ~ true
+
+            assert(record[String]("x") == "str")
+            assert(record[Int]("x") == 42)
+            assert(record[Boolean]("x") == true)
+        }
+    }
+
+    "Tag derivation" - {
+        "derives tags for record" in pendingUntilFixed {
+            val record = "name" ~ "Alice" & "age" ~ 30
+            assertCompiles("""summon[Tag[Record["name" ~ String & "age" ~ Int]]]""")
+        }
+
+        "doesn't limit the use of nested records" in pendingUntilFixed {
+            val inner = "x" ~ 1 & "y" ~ 2
+            assertCompiles(""""z" ~ inner""")
+        }
+    }
+
+    "Render" - {
+        "simple record" in pendingUntilFixed {
```

Comment:
I couldn't find a way to collect the `Render` evidences for each of the field types